pkg/sdk/poc/generator/templates.go

```diff
@@ -3,186 +3,289 @@ package generator
 import "text/template"
 
 var PackageTemplate, _ = template.New("packageTemplate").Parse(`
-package {{.}}
+package {{ . }}
 `)
 
-var InterfaceTemplate, _ = template.New("interfaceTemplate").Parse(`
+var InterfaceTemplate, _ = template.New("interfaceTemplate").
+	Funcs(template.FuncMap{
+		"deref": func(p *DescriptionMappingKind) string { return string(*p) },
+	}).
+	Parse(`
 import "context"
 
-type {{.Name}} interface {
-	{{- range .Operations}}
-		{{.Name}}(ctx context.Context, request *{{.OptsField.DtoDecl}}) error
-	{{- end}}
+type {{ .Name }} interface {
+	{{- range .Operations }}
+		{{- if and (eq .Name "Show") .ShowMapping }}
+			{{ .Name }}(ctx context.Context, request *{{ .OptsField.DtoDecl }}) ([]{{ .ShowMapping.To.Name }}, error)
+		{{- else if and (eq .Name "Describe") .DescribeMapping }}
+			{{ if .DescribeKind }}
+				{{ if eq (deref .DescribeKind) "single_value" }}
+					{{ .Name }}(ctx context.Context, id {{ .ObjectInterface.IdentifierKind }}) (*{{ .DescribeMapping.To.Name }}, error)
+				{{ else if eq (deref .DescribeKind) "slice" }}
+					{{ .Name }}(ctx context.Context, id {{ .ObjectInterface.IdentifierKind }}) ([]{{ .DescribeMapping.To.Name }}, error)
+				{{ end }}
+			{{ end }}
+		{{ else }}
+			{{ .Name }}(ctx context.Context, request *{{ .OptsField.DtoDecl }}) error
+		{{- end -}}
+	{{ end }}
 }
 `)
 
 var OptionsTemplate, _ = template.New("optionsTemplate").Parse(`
-// {{.OptsField.KindNoPtr}} is based on {{.Doc}}.
-type {{.OptsField.KindNoPtr}} struct {
-	{{- range .OptsField.Fields}}
-			{{.Name}} {{.Kind}} {{.TagsPrintable}}
-	{{- end}}
+// {{ .OptsField.KindNoPtr }} is based on {{ .Doc }}.
+type {{ .OptsField.KindNoPtr }} struct {
+	{{- range .OptsField.Fields }}
+			{{ .Name }} {{ .Kind }} {{ .TagsPrintable }}
+	{{- end }}
 }
 `)
 
 // TODO: merge with template above? (requires moving Doc to field)
 var StructTemplate, _ = template.New("structTemplate").Parse(`
-type {{.KindNoPtr}} struct {
-	{{- range .Fields}}
-			{{.Name}} {{.Kind}} {{.TagsPrintable}}
-	{{- end}}
+type {{ .KindNoPtr }} struct {
+	{{- range .Fields }}
+			{{ .Name }} {{ .Kind }} {{ .TagsPrintable }}
+	{{- end }}
 }
 `)
 
 var DtoTemplate, _ = template.New("dtoTemplate").Parse(`
-{{define "DTO_STRUCT"}}
-type {{.DtoDecl}} struct {
-	{{- range .Fields}}
-		{{- if .ShouldBeInDto}}
-		{{.Name}} {{.DtoKind}} {{if .Required}}// required{{end}}
-		{{- end}}
-	{{- end}}
-}
-
-{{- range .Fields}}
-	{{if .IsStruct}}
-	{{template "DTO_STRUCT" .}}
-	{{end}}
-{{- end}}
-{{end}}
-
-//go:generate go run ../../dto-builder-generator/main.go
+//go:generate go run ./dto-builder-generator/main.go
```

Comment:
this should be parametrised, because now example won't work (I mentioned it in another comment).

You can add it as essential.